src/physics.cpp

```diff
@@ -209,6 +209,59 @@ void create_fission_sites(Particle& p, int i_nuclide, const Reaction& rx)
     // Sample delayed group and angle/energy for fission reaction
     sample_fission_neutron(i_nuclide, rx, &site, p);
 
+    // Iterated Fission Probability (IFP) method
+    // Needs to be done after the delayed group is found
+
+    vector<int> updated_ifp_delayed_groups;
+    vector<double> updated_ifp_lifetimes;
```

Comment:
This will result in memory allocation in the middle of a particle's transport, which is never desirable 1) because it can degrade threading efficiency and 2) it doesn't fit GPU programming models well. Is there any way we can avoid the memory allocation?

Reply:
With the refactoring, these vectors are only declared inside the "if (settings::ifp)" loop now. This should no longer impact the non-IFP user.

For the IFP context, I need to use vectors in the current implementation as the number of generations is only known at runtime (user input). There are options to use arrays instead with some limitations but this is not what I chose for this implementation.